backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/card/BpkCard.kt

```diff
@@ -0,0 +1,74 @@
+package net.skyscanner.backpack.compose.card
+
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.interaction.collectIsFocusedAsState
+import androidx.compose.foundation.interaction.collectIsPressedAsState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.semantics.Role
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkBorderRadius
+import net.skyscanner.backpack.compose.tokens.BpkElevation
+
+enum class BpkCardCorner {
+  Small,
+  Large,
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun BpkCard(
+  modifier: Modifier = Modifier,
+  corner: BpkCardCorner = BpkCardCorner.Small,
+  onClick: (() -> Unit)? = null,
+  interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+  onClickLabel: String? = null,
+  role: Role? = null,
+  content: @Composable () -> Unit,
+) {
+
+  val focused by interactionSource.collectIsFocusedAsState()
+  val pressed by interactionSource.collectIsPressedAsState()
+  val elevated = focused || pressed
+
+  val elevation by animateDpAsState(
+    when {
+      elevated -> BpkElevation.Xl
+      else -> BpkElevation.Base
+    }
+  )
+
+  val backgroundColor by animateColorAsState(
+    when {
+      elevated -> BpkTheme.colors.backgroundElevation02
+      else -> BpkTheme.colors.backgroundElevation01
+    }
+  )
+
+  val size = when (corner) {
+    BpkCardCorner.Small -> BpkBorderRadius.Md
+    BpkCardCorner.Large -> BpkBorderRadius.Lg
+  }
+
+  Card(
```

Comment:
Looking at the material card there's two separate card implementations depending on if there's click support vs not. looking at surface the `clickAndSemanticsModifier` is quite different depending on if it is or not. i suspect there might be subtle differences in the behaviour and the difference is there for a reason - especially around accessibility. have you tested with talkback etc?

Reply:
If you check out the `Card` implementation code, it reffers to `Surface`, which have `clickAndSemanticsModifier` you've mentioned. This modifier is set to be standart `Modifier.clickable`, so the behaviour is practically the same. Is difference is in the modifiyers order (and that's why its extracted as an argument): the click modifiiyer is being applied only after shadow/background/clip modifiers.

The reason for that is basically the same reason why the overload with onClick lambda exists: if you just apply `Modifier.clickable` to default Card modifier, the ripple from the selection won't be clip according to the card shape – as it's being applied befire the clip.

So, it's should be exactly the same semantically – and even more precise since shadow should also not be in a screen reader selector indicator :D

Reply:
Yeah so there's two `Surface` implementations - one is `Modifier.clickable` - the other is `Modifier.semantics` the big difference is that one has `mergeDescendants` set to true (the clickable) vs false, which might mean that if there's clickable items within the card it might not behave correctly as it's treated as one card, rather than just a container. I believe this is similar to atomic vs non-atomic where it either gets treated as one item or multiple items.

Reply:
I see. I would rather not have 2 diffrerent functions but rather used different card callds depending on whether onClick is null – what do you think?

Reply:
Given that the non-clickable version doesn't have a lot of the properties in the clickable version it would be very confusing if you could set them, but they wouldn't do anything, so two different functions are probably better